java/ql/lib/semmle/code/java/regex/regex.qll

```diff
@@ -0,0 +1,933 @@
+import java
```

Comment:
The ordering of the member predicates of `RegexString` seems a bit arbitrary. Could we topologically sort them in dependency order? I think that would make the code easier to read and any potential recursion would be easier to recognise.

Reply:
There's still `escapedCharacter`, which is used before being defined.

Reply:
Agree. I just found that I couldn't review this code efficiently without it being topologically sorted. So we should sort the others similarly as a follow-up.